vault/logical_system.go

```diff
@@ -1717,6 +1733,159 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 		}
 	}
 
+	// user-lockout config
+	{
+		var apiuserLockoutConfig APIUserLockoutConfig
+
+		userLockoutConfigMap := data.Get("user_lockout_config").(map[string]interface{})
+		var err error
+		// Augmenting userLockoutConfigMap for some config options to treat them as comma separated entries
+		err = expandStringValsWithCommas(userLockoutConfigMap)
+		if err != nil {
+			return logical.ErrorResponse(
+					"unable to parse given auth user lockout config information"),
+				logical.ErrInvalidRequest
+		}
+		if userLockoutConfigMap != nil && len(userLockoutConfigMap) != 0 {
+			err := mapstructure.Decode(userLockoutConfigMap, &apiuserLockoutConfig)
+			if err != nil {
+				return logical.ErrorResponse(
+						"unable to convert given user lockout config information"),
+					logical.ErrInvalidRequest
+			}
+
+			switch strings.ToLower(mountEntry.Type) {
+			case "all", "ldap", "approle", "userpass":
```

Comment:
is there an "all" mount type? 
On a side note, it seems that only "ldap", "approle", "userpass" are supported. If that is the case, it would be good to document what auth mounts `all` refers to.

Reply:
"all" is used to set lockout configuration to all the supported auth types (ldap, userpass and approle).  I will add this as a comment here and will also add it in the documentation changes. Thanks

Reply:
Yes, mountEntry.Type cannot be "all", I will remove "all" from the list.